### PR TITLE
Create function-app-deploy.yml

### PR DESCRIPTION
**Pull request by albal**
Create a new action to deploy the function app to dev

### CODE REVIEW

File: .github/workflows/function-app-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    name: 'Function App Deploy'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Sign in to Azure 
+        uses: azure/login@v1
+        with:
+          creds: '{"clientId":"${{ secrets.FAD_CLIENT_ID }}","clientSecret":"${{ secrets.FAD_CLIENT_SECRET }}","subscriptionId":"${{ secrets.FAD_SUBSCRIPTION_ID }}","tenantId":"${{ secrets.FAD_TENANT_ID }}"}'
+
+      - name: Deploy Function App (Dev)
+        run:  az functionapp deployment source config --branch main --manual-integration --name s185d01-search --repo-url https://github.com/DFE-Digital/childrens-social-care-cpd-indexer --resource-group s185d01-childrens-social-care-cpd-rg
```
Comment (killij) [inline .github/workflows/function-app-deploy.yml:18]: The branch will need to be different depending on the target env.

Comment (killij) [reply .github/workflows/function-app-deploy.yml:18]: Also the name and resource group params are hard coded here?

**Review by killij — CHANGES_REQUESTED** [review]

File: .github/workflows/function-app-deploy.yml (hunk 1 above)
Comment (albal) [reply .github/workflows/function-app-deploy.yml]: Code has been updated to use parameters and also user can select environment.

**Review by killij — APPROVED** [review]
LG

**Pull request merged**